### ISSUE TITLE
Flathub published a version (2.10.34) early and it has a known text recoloring bug - Revert to 2.10.32

### ISSUE THREAD

**Issue by LinuxBeaver** [flathub/org.gimp.GIMP#196]
Gimp's team has NOT released 2.10.34 it is not ready and has a known text recoloring bug.

https://discourse.flathub.org/t/someone-put-an-unstable-version-of-gimp-2-10-on-flathub-needs-reversing/3768

https://www.reddit.com/r/GIMP/comments/11ak04q/gimp_21034_only_types_in_white/

Please revert to Gimp 2.10.34 now.

**Comment by Jehan** [comment]
> Gimp's team has NOT released 2.10.34 it is not ready and has a known text recoloring bug.

This package is managed by the GIMP team. So yes, we have released GIMP 2.10.34 (i.e. the code was tagged), simply we have not published the news yet, because we are still waiting for a bit more binaries to be built (at least macOS or Windows ones). It's common that binaries don't come all at the same time, especially as we are relying on volunteers able to free up their own personal time.

> known text recoloring bug.

You say it's a "known" bug. So I was hoping to find some clue in your links about maybe a known bug in Pango or Cairo or whatnot. But I don't see anything. Why are you saying it's a non-bug? Your links seem to indicate the opposite so far (i.e. that it's a new bug).

> Please revert to Gimp 2.10.34 now.

You probably mean 2.10.32. Anyway this probably won't do much. We need to know where the bug comes from. Otherwise all it may achieve is having the same issue with 2.10.32.

**Comment by Jehan** [comment]
Ok so I reverted the Cairo dependency bump in #197, where the issue was coming from (as far as my tests went). This should be into effect in a few hours. Please test and comment.

You may also try this test build:

```
flatpak install --user https://dl.flathub.org/build-repo/7564/org.gimp.GIMP.flatpakref
```

**Comment by LinuxBeaver** [comment]
How do I launch this version of Gimp to test it?

**Comment by Jehan** [comment]
Uninstall the flatpak GIMP first, then in a terminal, run the command line above (which will likely ask you to confirm with a 'y'; so confirm). Then run the flatpak normally.

**Comment by LinuxBeaver** [comment]
The 2.10.34 update made things worse. It now has artifact around the outlines.


How do I role back to 2.10.32 so I can work?

> flatpak update --commit=7fad12864e7741e49d104e6e35ac36e63f0d08f165ac94d22f773624499ff3ed org.gimp.GIMP

should work but it doesn't. Please help me if you can.

**Comment by TingPing** [comment]
> flatpak update --commit=7fad12864e7741e49d104e6e35ac36e63f0d08f165ac94d22f773624499ff3ed org.gimp.GIMP

FWIW this works for me.

**Comment by LinuxBeaver** [comment]
- it doesn't work for me and i am losing my cool. I have work to do with GImp and either 2.10.34 gets fixed or I figure out how to downgrade.

**Comment by LinuxBeaver** [comment]
Is there a way to install 2.10.32 similar to the 

> flatpak install --user https://dl.flathub.org/build-repo/

command? I want to see if that will work.

**Comment by Jehan** [comment]
This will update your current flatpak exactly to the same version as existed before the update:
```
flatpak update -y --user --commit=206e51f29170e2852001fd12af41bd42bdf4322cb6d55ed0330774ceab6b2f07 org.gimp.GIMP//stable
```

Also for the record, in GIMP repo, there is a tool to automatize listing previous flathub builds and installing them:

```
$ tools/flatpak-releases 
 0: Revert "Update Cairo to version 1.17.8." (303a64aa) [2023-02-25 11:51:00 +0000] - app-commit: f5281d0637b84e3ed219bfd0a312d8d0ff603a60809604373c8923b62e8ee3d5
 1: Patch Changes in version 2.10.34 (2fdc5b30) [2023-02-24 14:50:38 +0000] - app-commit: dcd1ec70ecefad72290cec53cb1157d4c23d4b1cb7d0915d34e48ab6a9cfaf6d
 2: Releasing GIMP 2.10.34. (1955da94) [2023-02-23 00:13:59 +0000] - app-commit: f4e6e8c75ce77d69f28e9a3d7192b97193a3700c3b67b053483909c141b59935
 3: Bump libheif version from 1.13.0 to 1.14.0. (9d9431a4) [2022-11-24 18:08:37 +0000] - app-commit: 206e51f29170e2852001fd12af41bd42bdf4322cb6d55ed0330774ceab6b2f07
 4: Bump back to GNOME runtime version 43. (84ce16aa) [2022-11-16 18:29:17 +0000] - app-commit: 7fad12864e7741e49d104e6e35ac36e63f0d08f165ac94d22f773624499ff3ed
 5: Update babl and GEGL to last releases. (d57f1b3d) [2022-11-14 15:08:31 +0000] - app-commit: a447ad827275c304f122545af1e326f94858db3f01b3595f1374e28c948870bd
 6: Update 2 modules (dd91153a) [2022-11-12 21:38:16 +0000] - app-commit: decae5425cab6312febddfb2714c5f5d748edff5d903ffc7363b5a8985d07077
 7: Remove lcms too. (18abfd1d) [2022-11-12 19:23:32 +0000] - app-commit: 0fb43c0eab0a7a93b276e2f3abdc5051dcf64ce562b495608af50126626a0bbc
$ tools/flatpak-releases -3
[1/2] Installing org.gimp.GIMP
Looking for matches…
Skipping: org.gimp.GIMP/x86_64/stable is already installed
[2/2] Updating to commit '206e51f29170e2852001fd12af41bd42bdf4322cb6d55ed0330774ceab6b2f07' (3's previous build)
Looking for updates…


        ID                    Branch         Op         Remote         Download
 1. [✓] org.gimp.GIMP         stable         u          flathub        70.4 MB / 126.8 MB

Warning: Not exporting file org.gimp.GIMP.appdata.xml of unsupported type.
Updates complete.
Build 3 released on 2022-11-24 18:08:37 +0000 was installed.
Build subject: Bump libheif version from 1.13.0 to 1.14.0. (9d9431a4)
Build commit on flathub: 206e51f29170e2852001fd12af41bd42bdf4322cb6d55ed0330774ceab6b2f07
```

This tool also works for betas and nightlies with `--beta` and `--nightly` options.

**Comment by LinuxBeaver** [comment]
it worked things are back to normal. I'd like to learn how to use that tool that Gimp provides to downgrade flatpaks.

**Comment by Jehan** [comment]
It's just a shell script so even if you don't build GIMP or anything, you could just copy the script somewhere and use it when needed: https://gitlab.gnome.org/GNOME/gimp/-/blob/master/tools/flatpak-releases

I made it for development usage. It's useful for instance when we want to bisect issues or match debugger traces offset to accurate file lines knowing the exact flatpak version used by someone (as the flatpak build commit is in the `gimp -v` output).

**Comment by LinuxBeaver** [comment]
I ran the script for Gimp 2.10.30 and this happened. 

[contrast@cosmos ~]$ /home/contrast/Desktop/gimp-GIMP_2_10_30/build/flatpak/gimp-flatpak-cron.sh 
Usage: /home/contrast/Desktop/gimp-GIMP_2_10_30/build/flatpak/gimp-flatpak-cron.sh <GPKKEY> <REPO> <MAIL>

    GPKKEY: GPG Key ID to sign flatpak commits with
    REPO: local path to flatpak repository directory
    MAIL: email address to send report to
[contrast@cosmos ~]$ 


What do I do?

**Comment by Jehan** [comment]
@LinuxBeaver That's the wrong script. The one you try to run is some old script which I should probably delete actually (back from the early times of flatpak when we were planning on building it in a cron on a server, before the CI and Flathub times). Click the link I gave above instead. Then the "open raw" button and download.

Or I give you directly the raw link: https://gitlab.gnome.org/GNOME/gimp/-/raw/master/tools/flatpak-releases

**Comment by LinuxBeaver** [comment]
So how would I get 2.10.30 with this? As from what I can tell Flathub doesn't list 2.10.30 anymore.

**Comment by Jehan** [comment]
Well if it's not in the history, then it's not possible. Flathub used to keep a [dozen older builds](https://www.gimp.org/news/2022/02/25/gimp-2-99-10-released/#tool-to-bisect-test-with-older-versions-of-flatpak) but it seems they may have lessened the history cache (or it's date-based or whatnot?).

**Comment by Jehan** [comment]
Fixed by #199.

**Comment by LinuxBeaver** [comment]
I still have this problem in 2.10.34 but oh well.
